### DISCUSSION TITLE
Typescript error when setting export const experimental_ppr = true;

### DISCUSSION THREAD

**Discussion by ryanknight09** [vercel/next.js#70735] [Help]
### Summary

When enableing ppr in Next.js 15RC, the export const experimental_ppr = true; in a page.tsx or layout.tsx file is considered an error by typescript still. 

`"experimental_ppr" is not a valid Next.js entry export value.ts(71002)`

Is there a fix to this or am I missing something? The feature works as expected and the builds come work just fine as well. The type error is just annoying in my layout file. Thanks for the insight. 

### Additional information

Next config:

`
/** @type {import('next').NextConfig} */
const nextConfig = {
    experimental: {
        ppr: 'incremental',
    },
};

export default nextConfig;
`

**Comment by icyJoseph** [comment]
Hi,

I think the RC branch hasn't been updated for a while now.

I found this, merged to canary, PR: 

- https://github.com/vercel/next.js/pull/66141

Which shows, that the issue you are facing has been fixed, in canary. My guess is that RC hasn't received these commits just yet.

I see in the TypeScript plugin code, that `experimental_ppr`, is indeed allowed.

Also, this comment, https://github.com/vercel/next.js/discussions/70718#discussioncomment-10823844, validates that the RC, hasn't been updated for some time.

**Reply by icyJoseph** [reply]
And btw, I think, you can put a `// @ts-expect-error ppr not in RC types` kind of comment above `export const experimental_ppr = true` to allow the build to go on.

**Reply by adrianbravomr** [reply]
I'm in Next.js 15.0.0-canary.56 and having the same issue.

**Reply by icyJoseph** [reply]
That's a 3 months old canary, chances are that you need a later version.

**Reply by adrianbravomr** [reply]
Thanks @icyJoseph 
I'm following 'Learn Next Js' course from Vercel, and noticed it is using canary.56 (An old version like you said), i've update it to canary.179 with same results, i'm doing additional research trying to find the issue.

**Reply by icyJoseph** [reply]
Aha, that's odd though 🤔 maybe that PR is not released into that canary (for v15)

**Reply by icyJoseph** [reply]
I see that the fix is still in the canary git branch: 
- https://github.com/vercel/next.js/blob/canary/packages%2Fnext%2Fsrc%2Fserver%2Ftypescript%2Frules%2Fconfig.ts

API_DOCS has experimental_ppr and that is checked before emitting the TS error.

**Reply by adrianbravomr** [reply]
At the end, for me, it was IntelliCode VS Code Extension screwing things up. I've disabled it until i find a better alternative (And learn to perform a more efficient debugging).

Thanks for the support @icyJoseph

**Reply by ryanknight09** [reply]
Ah, thanks @icyJoseph.

**Reply by icyJoseph** [reply]
If this worked out for you, please don't forget to mark as answered 👍
